types/money/index.d.ts

```diff
@@ -0,0 +1,87 @@
+// Type definitions for money 0.2
+// Project: https://github.com/openexchangerates/money.js
+// Definitions by: Ivan Ergunov <https://github.com/hozblok>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+declare namespace fx {
+    type Currency = string;
+
+    type Rates = Record<string, number | string>;
+
+    type Options = {
+        from?: Currency | null;
+        to?: Currency | null;
+    } | null;
+
+    type Value = string | number | string[] | number[];
+
+    type Result<T_VAL extends Value> = T_VAL extends string | number ? number : number[];
+
+    interface WrapperPrototype<T_VAL extends Value> {
+        convert: {
+            (opts?: Options): Result<T_VAL>;
+        };
+
+        from: {
+            (currency: Currency): Wrapper<T_VAL>;
+        };
+
+        to: {
+            (currency: Currency): Result<T_VAL>;
+        };
+    }
+
+    interface Wrapper<T_VAL extends Value> extends WrapperPrototype<T_VAL> {
+        readonly _v: number;
+        readonly _fx?: Currency;
```

Comment:
those are private, what's the purpose when consuming this types?

Reply:
I viewed it as protected not as private. The case is the user want to inherit the NewableFunction `fx`:

```typescript
import fx = require('money');

class MyFX<T_VAL extends fx.Value> extends fx<T_VAL> {
    constructor(val: T_VAL) {
        super(val);
    }

    getValue() {
        // ... do something
        return this._v // number
    }
}
```

Maybe remove `readonly`... How do you think?

Reply:
don't be mislead by the power of types. This is a module and I think it only exports a single symbol, which is an instance of those constructs. There is no way for user to extend something, am I seeing this right?

Reply:
Oh, you are right. Sorry for misunderstanding.